app/models/compute_resources/foreman/model/ec2.rb

```diff
@@ -51,6 +51,7 @@ def create_vm(args = { })
       args[:groups].reject!(&:empty?) if args.has_key?(:groups)
       args[:security_group_ids].reject!(&:empty?) if args.has_key?(:security_group_ids)
       args[:associate_public_ip] = subnet_implies_is_vpc?(args) && args[:managed_ip] == 'public'
+      args[:private_ip_address] = args[:interfaces_attributes][:"0"][:ip]
```

Comment:
I tested that as well. If you don't set and IP address in the host interface amazon will assign one free from the Subnet selected  for that VPC instance because private_ip_address will have no value.